### PR TITLE
PWGHF: Add Ds task

### PR DESCRIPTION
**Pull request by stefanopolitano**
- Add Ds task in config_task.sh
- Add Ds configuration in dpl-config_run3/run5_hf.json
- Add Ds workflow
- Related PR: https://github.com/AliceO2Group/O2Physics/pull/941

### CODE REVIEW

File: codeHF/dpl-config_run3.json [hunk 1]
```diff
@@ -513,6 +546,38 @@
...
+                "4.",
+                "5.",
+                "6.",
+                "8.",
+                "12.",
+                "16."
+            ]
+        }
+    },
+    "hf-task-ds-mc": {
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:565]: This part should disappear once the MC part becomes a process function.

File: codeHF/config_tasks.sh [hunk 2]
```diff
@@ -119,7 +122,7 @@ function Clean {
 function AdjustJson {
   # Make a copy of the default JSON file to modify it.
   JSON_EDIT=""
-  if [[ $APPLYCUTS_D0 -eq 1 || $APPLYCUTS_DPLUS -eq 1 || $APPLYCUTS_LC -eq 1 || $APPLYCUTS_XIC -eq 1 || $APPLYCUTS_JPSI -eq 1 || $APPLYCUTS_LCK0SP -eq 1 || $APPLYCUTS_X -eq 1 || $APPLYCUTS_CHIC -eq 1 || $APPLYCUTS_XICC -eq 1 ]]; then
+  if [[ $APPLYCUTS_D0 -eq 1 || $APPLYCUTS_DPLUS -eq 1 || $APPLYCUTS_DS -eq 1 || $APPLYCUTS_LC -eq 1 || $APPLYCUTS_XIC -eq 1 || $APPLYCUTS_JPSI -eq 1 || $APPLYCUTS_LCK0SP -eq 1 || $APPLYCUTS_X -eq 1 || $APPLYCUTS_XICC -eq 1 ]]; then
```
Comment (vkucera) [inline codeHF/config_tasks.sh:125]: Seems like you removed `APPLYCUTS_CHIC`.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Please link the O2 PR.

File: codeHF/dpl-config_run3.json (hunk 1 above)
Comment (stefanopolitano) [reply codeHF/dpl-config_run3.json:565]: Hi @vkucera! Ok, I took the D+ as an example and it still has this MC part. Do you want me to remove it anyway?

Comment (vkucera) [reply codeHF/dpl-config_run3.json:565]: Yes, the change from workflow options to process functions should be propagated everywhere.

**Comment by stefanopolitano** [comment]
Here below is linked the O2 PR for the Ds task:
 O2 PR - Ds Task: _https://github.com/AliceO2Group/O2Physics/pull/96_

**Comment by vkucera** [comment]
This requires some updates.

**Comment by vkucera** [comment]
Please update

File: codeHF/config_tasks.sh [hunk 3]
```diff
@@ -222,6 +225,12 @@ function AdjustJson {
     ReplaceString "\"d_selectionFlagD0bar\": \"0\"" "\"d_selectionFlagD0bar\": \"1\"" "$JSON" || ErrExit "Failed to edit $JSON."
   fi
 
+  # Enable Ds selection.
+  if [ $APPLYCUTS_Ds -eq 1 ]; then
```
Comment (vkucera) [inline codeHF/config_tasks.sh:229]:
```suggestion
  if [ $APPLYCUTS_DS -eq 1 ]; then
```

File: codeHF/config_tasks.sh [hunk 4]
```diff
@@ -431,6 +443,7 @@ function MakeScriptPostprocess {
   [[ $DOO2 -eq 1 && $ISMC -eq 1 ]] && {
     PARTICLES=""
     [ $DOO2_TASK_D0 -eq 1 ] && PARTICLES+=" d0 "
+    [ $DOO2_TASK_Ds -eq 1 ] && PARTICLES+=" ds "
```
Comment (vkucera) [inline codeHF/config_tasks.sh:446]:
```suggestion
    [ $DOO2_TASK_DS -eq 1 ] && PARTICLES+=" ds "
```

File: codeHF/Compare.C [hunk 5]
```diff
@@ -87,6 +87,12 @@ Int_t Compare(TString filerun3 = "AnalysisResults_O2.root", TString filerun2 = "
   AddHistogram(vecHisD0MC, labelParticle + ", matched non-prompt: #it{p}_{T}^{rec} (GeV/#it{c})", "hPtRecoFeeddwD0Kpi", "hf-task-d0/hPtRecSigNonPrompt", 2, 1, 0);
   AddHistogram(vecHisD0MC, labelParticle + ", gen. non-prompt: #it{p}_{T}^{gen} (GeV/#it{c})", "hPtGenLimAccFeeddwD0Kpi", "hf-task-d0/hPtGenNonPrompt", 2, 1, 0);
 
+  VecSpecHis vecHisDs;
+  AddHistogram(vecHisDs, "#it{p}_{T} D_{s}^{#pm} (GeV/#it{c})", "hPtDs", "hf-task-ds/hPt", 2, 1, 0);
+  AddHistogram(vecHisDs, "3-prong mass (K K #pi) (GeV/#it{c}^{2})", "hInvMassDs", "hf-task-ds/hInvMass", 5, 0, 0);
```
Comment (vkucera) [inline codeHF/Compare.C:92]:
```suggestion
  AddHistogram(vecHisDs, "3-prong mass (K K #pi) (GeV/#it{c}^{2})", "hInvMassDs", "hf-task-ds/hMass", 5, 0, 0);
```

**Review by vkucera — CHANGES_REQUESTED** [review]
I spotted a few typos. Please double-check that there are no more.

**Comment by stefanopolitano** [comment]
Hi @vkucera ! Thanks for the review, I had a look and I could not find any more typos

**Comment by vkucera** [comment]
> Hi @vkucera ! Thanks for the review, I had a look and I could not find any more typos

Good, thanks for the quick checks.

**Review by vkucera — APPROVED** [review]

**Pull request merged**